packages/datetime/src/dateInput.tsx

```diff
@@ -447,10 +464,37 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
             // the page. tabbing forward should *not* close the popover, because
             // focus will be moving into the popover itself.
             this.setState({ isOpen: false });
+        } else if (e.which === Keys.ESCAPE) {
+            this.setState({ isOpen: false });
+            this.inputRef.blur();
         }
         this.safeInvokeInputProp("onKeyDown", e);
     };
 
+    private registerPopoverBlurHandler = () => {
+        const node = ReactDOM.findDOMNode(this);
+        const tabbableElements = node.querySelectorAll("[tabindex]:not([tabindex='-1'])");
```

Comment:
any sense in supporting other focusable things like `<button>` or are we safe cuz the calendar contents are well-defined and only use `tabIndex`? (hoping we're safe... this is a rabbit hole)

Reply:
- [ ] check to see if we can reasonably support other focusable components

Reply:
Actually, we will need to if we also want to support datetime.